estate/models/estate_offer.py

```diff
@@ -0,0 +1,59 @@
+from odoo import api, fields, models
+from odoo.tools import relativedelta
+from odoo.exceptions import UserError
+
+from .. import util
+
+class OfferEstate(models.Model):
+    _name = "estate.property.offer"
+    _description = "An offer for a property"
+
+    _sql_constraints = [
+        ("check_price", "check(price > 0)", "An offer's price must be strictly positive"),
+    ]
+
+    price = fields.Float("Price")
+    status = fields.Selection(string="Status", copy=False, selection=[
+        ("accepted", "Accepted"),
+        ("refused", "Refused"),
+    ])
+
+    partner_id = fields.Many2one("res.partner", string="Partner", required=True)
+    property_id = fields.Many2one("estate.property", string="Property", required=True)
+
+    validity = fields.Integer("Validity", default=7)
+    date_deadline = fields.Date("Deadline", compute="_compute_date_deadline", inverse="_inverse_date_deadline")
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = record.create_date.date() + relativedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            if record.date_deadline:
+                record.validity = (record.date_deadline - record.create_date.date()).days
+
+    @util.action
+    def action_set_accepted(self):
+        for record in self:
+            if record.status == "accepted":
+                continue
+
+            property = record.property_id
+
+            # Only one offer can be accepted
+            for status in property.offer_ids.mapped("status"):
+                if status == "accepted":
+                    raise UserError("Only one offer can be accepted")
+
+            property.selling_price = record.price
+            record.status = "accepted"
+
+    @util.action
+    def action_set_refused(self):
+        for record in self:
+            if record.status == "accepted":
+                record.property_id.selling_price = 0.
+
+            record.status = "refused"
```

Comment:
This is fine but you may see the part 2 of [my comment](https://github.com/odoo/tutorials/pull/43#discussion_r1531754577) interesting.